oak_functions/loader/src/server.rs

```diff
@@ -592,14 +592,13 @@ fn check_export_function_signature(
 /// response may be padded by a number of trailing 0s before encoding the response as a binary
 /// protobuf message. In this case, the `length` in the response will contain the effective length
 /// of the `body`. This response is guaranteed to comply with the policy's size restriction.
-pub async fn apply_policy<F, S>(policy: ServerPolicy, function: F) -> anyhow::Result<Response>
+pub async fn apply_policy<F>(policy: ServerPolicy, function: F) -> anyhow::Result<Response>
```

Comment:
Uhmm I'm not super convinced though. For instance this function is async, but it expects the underlying function to be sync, which means that e.g. if there is a sleep or some other blocking IO this async task will be stuck in there. Also ideally even the Wasm engine itself would be async, in which case we would have to undo this change, right? That said, please go ahead with this change if you think it's useful (and while our wasm engine of choice is sync anyways).

Reply:
> Uhmm I'm not super convinced though. For instance this function is async, but it expects the underlying function to be sync, which means that e.g. if there is a sleep or some other blocking IO this async task will be stuck in there.

I'm not sure I follow, why does the async status of the accepted function make a difference? It's still run inside the async task spawned by this function: `let task = tokio::spawn(async move { function() });`

Reply:
I don't think there are any async Wasm engines available. The only place where async things could possibly make sense would be inside an ABI call, but I don't know how a Wasm engine would implement that. Even if it was supported I think we should always try to avoid async in extensions/ABI calls, as these are only useful for the types of things we want to avoid (e.g. disk or network access).

Reply:
Reading about this a bit more, out of curiosity, how we considered using tokio's `spawn_blocking` instead of the normal `spawn` here? It seems like it's intended for the kind of thing we're doing. https://docs.rs/tokio/0.2.25/tokio/task/fn.spawn_blocking.html

Reply:
I (for one) was not aware of it, from reading the Rust doc I agree it seems sensible

Reply:
Will open a PR to use it then. :D